drv/gimlet-hf-server/src/main.rs

```diff
@@ -157,12 +165,17 @@ fn main() -> ! {
     let mut buffer = [0; idl::INCOMING_SIZE];
     let mut server = ServerImpl {
         qspi,
-        block: [0; 256],
         capacity: 1 << log2_capacity,
         mux_state: HfMuxState::SP,
         dev_state: HfDevSelect::Flash0,
         mux_select_pin: cfg.sp_host_mux_select,
         dev_select_pin: cfg.flash_dev_select,
+        hash: HashData {
```

Comment:
Nit: this could be `HashData::new(HASH.get_task_id())`, since it's also initialized in `cosmo-hf`.